File: scharp/day/six/CSharpSpecFlowHealthcareExample/Domain/AuthenticationService.cs

namespace CSharpSpecFlowHealthcareExample.Domain
{
    public sealed class AuthenticationService
    {
        private readonly Dictionary<string, string> _users =
        new(StringComparer.OrdinalIgnoreCase);

        public void AddUser(
            string username,
            string password)
        {
            _users[username] = password;
        }

        public bool Login(
            string username,
            string password)
        {
            return _users.TryGetValue(
                       username,
                       out var savedPassword)
                   && savedPassword == password;
        }
    }
}

File: java/java-practise/src/main/java/day/six/java-cucumber-healthcare/src/main/java/domain/AuthenticationService.java
package domain;

import java.util.HashMap;
import java.util.Map;
import java.util.Objects;

public final class AuthenticationService {
    private final Map<String, String> users = new HashMap<>();

    public void addUser(
            String username,
            String password
    ) {
        users.put(
                Objects.requireNonNull(
                        username,
                        "username must not be null"
                ),
                Objects.requireNonNull(
                        password,
                        "password must not be null"
                )
        );
    }

    public boolean login(
            String username,
            String password
    ) {
        return users.containsKey(username)
                && Objects.equals(
                users.get(username),
                password
        );
    }
}
